crates/data_components/src/debezium/change_event.rs

```diff
@@ -21,7 +21,29 @@ use std::{
 
 use serde::{Deserialize, Serialize};
 
-/// A representation of a Debezium Change Event.
+/// A representation of a Debezium Change Event Key.
+#[derive(Serialize, Deserialize)]
+pub struct ChangeEventKey {
+    pub schema: Schema,
+    pub payload: serde_json::Value,
+}
+
+impl ChangeEventKey {
+    pub fn from_bytes(bytz: &[u8]) -> Result<Self, serde_json::Error> {
+        serde_json::from_slice(bytz)
+    }
+
+    #[must_use]
+    pub fn get_primary_key(&self) -> Vec<String> {
+        self.schema
+            .fields
+            .iter()
+            .filter_map(|field| field.field.clone())
```

Comment:
this is bit confusing without knowing the data construct here, may add a example here what the schema fields can be like?

Reply:
Added an example JSON payload